AsyncDisplayKit/ASViewController.mm

```diff
@@ -132,4 +142,70 @@ - (ASInterfaceState)interfaceState
   return _node.interfaceState;
 }
 
+#pragma mark - ASEnvironmentTraitCollection
+
+- (ASEnvironmentTraitCollection)displayTraitsForTraitCollection:(UITraitCollection *)traitCollection
```

Comment:
Might be a better opportunity to allow the developer to provide a subclass of ASTraitCollection (or ASEnvironmentTraitCollection) with extra properties that `displayTraitsContext` currently provide. For the common case, most people won't need to have `displayTraitsContext`, so figuring out a way to reduce API footprint would be a bonus. Are we constrained introducing a design like that?